AutoFoldCode.py

```diff
@@ -64,17 +110,36 @@ def run(self):
 # Restore the saved folds for the given view.
 def _restore_folds(view):
   settings = sublime.load_settings(__storage_file__)
-  for a, b in settings.get(view.file_name(), []):
-    view.fold(sublime.Region(a, b))
+  view_folds_data = settings.get(view.file_name(), {})
```

Comment:
Running this I had a few errors logged:

```
Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime_plugin.py", line 525, in on_load_async
    callback.on_load_async(v)
  File "/Users/acheronfail/Library/Application Support/Sublime Text 3/Packages/AutoFoldCode/AutoFoldCode.py", line 54, in on_load_async
    _restore_folds(view)
  File "/Users/acheronfail/Library/Application Support/Sublime Text 3/Packages/AutoFoldCode/AutoFoldCode.py", line 113, in _restore_folds
    view_folds_data = settings.get(view.file_name(), {})
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime.py", line 1167, in get
    return sublime_api.settings_get_default(self.settings_id, key, default)
TypeError: String required
Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime_plugin.py", line 525, in on_load_async
    callback.on_load_async(v)
  File "/Users/acheronfail/Library/Application Support/Sublime Text 3/Packages/AutoFoldCode/AutoFoldCode.py", line 54, in on_load_async
    _restore_folds(view)
  File "/Users/acheronfail/Library/Application Support/Sublime Text 3/Packages/AutoFoldCode/AutoFoldCode.py", line 113, in _restore_folds
    view_folds_data = settings.get(view.file_name(), {})
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime.py", line 1167, in get
    return sublime_api.settings_get_default(self.settings_id, key, default)
TypeError: String required
```

Might be a good idea to always check that `view.file_name()` exists beforehand:

```py
view.file_name() != None
```

Reply:
Thanks! Do you have any suggestions for workflows to test? I'm not familiar with the range of workflows - I guess this comes from something I didn't try (windows without name?)

Reply:
Yeah, this comes from using unsaved buffers and such. I'd make sure to test:

* open a file, create a fold, save the file (then close and re-open)
* open a file, create a fold, hot exit (then close and re-open)
* create a new buffer, create a fold, hot exit (then close and re-open)
* create a new buffer, create a fold, save the file (then close and re-open)

And then also check the commands:

* unfold all open files (should reset the storage for _all open files_)
* clear all folds (should reset the storage file)

Reply:
Done and tested everything, with an additional test of unfold the current file.